src/main/java/com/passbook/sparkeighteen/service/UserService.java

```diff
@@ -99,6 +103,42 @@ private Integer calculateAge(LocalDate dob) {
         return Period.between(dob, LocalDate.now()).getYears();
     }
 
+    public ProfileResponse updateProfile(Integer userID, @Valid ProfileRequest request) {
+        Optional<UserEntity> optionalUser = userRepository.findById(userID);
+        if (optionalUser.isEmpty()) {
+            return ProfileResponse.builder()
+                    .message("User ID is missing. Retry with registered user")
+                    .userID(userID)
+                    .build();
+        }
+
+        final UserEntity user = optionalUser.get();
+        Optional<ProfileEntity> optionalProfile = profileRepository.findByUser(user);
+        if (optionalProfile.isEmpty()) {
+            return ProfileResponse.builder()
+                    .message(String.format("Profile for user %s with UserID: %d is missing", user.getFirstname(), user.getId()))
+                    .build();
+        }
+
+        ProfileEntity profile = optionalProfile.get();
+
+        profile.setAadhar(request.getAadhar());
```

Comment:
If some one passes null aadhar card then we should be keeping already present aadhar card detail while saving 
do for all fields.

Reply:
Update changes, when edit profile you pass null then its take previous value, otherwise throw validation error